src/Common/MemoryBufferWriter.cs

```diff
@@ -307,5 +313,27 @@ protected override void Dispose(bool disposing)
                 Reset();
             }
         }
+
+        /// <summary>
+        /// Holds a byte[] from the pool and a size value. Basically a Memory but guaranteed to be backed by an ArrayPool byte[], so that we know we can return it.
+        /// </summary>
+        private struct BufferSegment
+        {
+            public byte[] Buffer { get; }
+            public int Length { get; }
+
+            public ReadOnlyMemory<byte> Memory => Buffer.AsMemory(0, Length);
+
+            public BufferSegment(byte[] buffer, int size)
+            {
+                Buffer = buffer;
+                Length = size;
```

Comment:
Kind of weird that names are different

Reply:
What about renaming BufferSegment to CompletedSegment? Length can be renamed to Used.

Reply:
It should just be Length, the names aren't too strange (look at memory and span).

Reply:
Derp, I just renamed the property and didn't catch the parameter.